packages/dds/tree/src/feature-libraries/flex-map-tree/mapTreeNode.ts

```diff
@@ -482,9 +482,31 @@ class EagerMapTreeSequenceField<T extends FlexAllowedTypes>
 	extends EagerMapTreeField<T>
 	implements FlexTreeSequenceField<T>
 {
-	public get editor(): SequenceFieldEditBuilder<ExclusiveMapTree[]> {
-		throw unsupportedUsageError("Editing an array");
-	}
+	public readonly editor: SequenceFieldEditBuilder<ExclusiveMapTree[]> = {
+		insert: (index, newContent) => {
+			for (let i = 0; i < newContent.length; i++) {
+				const c = newContent[i];
+				assert(c !== undefined, "Unexpected sparse array content");
+				nodeCache.get(c)?.adoptBy(this, index + i);
+			}
+			this.edit((mapTrees) => {
+				mapTrees.splice(index, 0, ...newContent);
```

Comment:
Unless `newContent` is has small-ish bounded size, this will create too many arguments. This is why we have `TreeArrayNode.spread(nodes)` for our own editing function. I don't know how to get around that with the standard JS methods. I guess you could do it in batches of - say - 256, but I'm hoping there's a better option.

Reply:
Addressed in [this commit](https://github.com/microsoft/FluidFramework/pull/22193/commits/4999e51e8f70873f3d6ad1371608c0881ddab7de) - let me know what you think.

Reply:
How important is it to use `splice` for smaller content arrays? Should we not just unconditionally use `concat` if content array size is a concern?

Reply:
Probably also good to add a test for the large content array case (if we keep the dual codepaths).

Reply:
With splice it is an edit of an existing array with 0 new arrays allocated. With concat it is three new arrays allocated and three arrays dropped for GC. Seems worth using splice for the 99.9% IMO.

There is a test added in the commit linked above.